src/org/jitsi/impl/neomedia/rtcp/termination/strategies/BasicRTCPTerminationStrategy.java

```diff
@@ -1175,17 +1142,129 @@ public void maybeReport()
             }
         }
 
-        private boolean timeToSendRTCPReport()
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public long getTimeUntilNextProcess()
         {
-            final long now = System.currentTimeMillis();
+            return (lastUpdateTimeMs + RTCP_INTERVAL_VIDEO_MS)
+                - System.currentTimeMillis();
+        }
+    }
 
-            if (now >= nextTimeToSendRTCP)
+    /**
+     * Keeps track of the CNAMEs of the RTP streams that we've seen.
+     */
+    static class CNAMERegistry
+        extends ConcurrentHashMap<Integer, byte[]>
+    {
+        /**
+         * @param inPacket
+         */
+        public void update(RTCPCompoundPacket inPacket)
+        {
+            // Update CNAMEs.
+            RTCPPacket[] rtcps;
+
+            if (inPacket == null
+                || (rtcps = inPacket.packets) == null
+                || rtcps.length == 0)
             {
-                nextTimeToSendRTCP = now + RTCP_INTERVAL_VIDEO_MS;
-                return true;
+                return;
             }
 
-            return false;
+            for (RTCPPacket rtcp : rtcps)
+            {
+                if (RTCPPacket.SDES != rtcp.type)
+                    continue;
+
+                RTCPSDESPacket sdes = (RTCPSDESPacket) rtcp;
+                RTCPSDES[] chunks = sdes.sdes;
+
+                if (chunks == null || chunks.length == 0)
+                    continue;
+
+                for (RTCPSDES chunk : chunks)
+                {
+                    RTCPSDESItem[] items = chunk.items;
+
+                    if (items == null || items.length == 0)
+                        continue;
+
+                    for (RTCPSDESItem item : items)
+                    {
+                        if (RTCPSDESItem.CNAME == item.type)
+                            put(chunk.ssrc, item.data);
+                    }
+                }
+            }
+        }
+    }
+
+    class RTCPPacketTransformerImpl
```

Comment:
1. I don't know that extracting RTPPacketTransformerImpl and RTCPPacketTransformerImpl made the source code better. Sure, indentation was removed which is an advantage. A disadvantage is that you had to add constructors.

2. Why are these 2 classes not private?